sqlglot/parser.py

```diff
@@ -484,6 +495,30 @@ def _parse_cache(self):
             expression=self._parse_cte(),
         )
 
+    def _parse_partition(self):
+        def parse_values():
```

Comment:
we can move this after line 507 so it doesn't parse most of the time

Reply:
Good call. Moved.